### ISSUE TITLE
Improved training tracking: compute metrics, save predictions

### ISSUE THREAD

**Issue by jdeschamps** [CAREamics/careamics#294]
## Problem

Currently, only the loss on validation images is reported during training. We should allow the possibility to compute metrics (scale invariant PSNR, MicroSSIM) if applicable, and save predictions on a validation image.

Ideally, this would be compatible with the three supported logger:
- `CSVLogger`: metrics should be reported.
- `WandB`: metrics and prediction should be saved.
- `TensorBoard`: same as `WandB`.

Finally, predictions could be saved to a user-defined folder.

## Potential solution

Firstly, we now have the capacity to predict during training, although the solution is pretty hacky: https://github.com/CAREamics/careamics/pull/266.

Secondly, PyTorch Lightning should have all the engineering required to save the metrics and the predictions in the relevant loggers:
- [Logging from a LightningModule](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.loggers.wandb.html)
- [Log media in WandB](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.loggers.wandb.html)
- [TensorBoard](https://tensorboardx.readthedocs.io/en/stable/tensorboard.html#tensorboardX.SummaryWriter)

## Linked issues

https://github.com/CAREamics/careamics/issues/299: better convenience functions for N2V
